### PR TITLE
62: add support for entities and entity_lists

### LINKED ISSUE #62: Support entities
- [x] decide whether we're using "dataset" or "entity list". The former is what the API calls a collection of entities, the latter is what's user-facing. `pyodk` is kind of in the middle. I'm leaning towards "entity list" because it matches form design and it's often the same person who design forms and does analysis. I think we should [align with ruODK](https://github.com/ropensci/ruODK/pull/153/files#r1522000629). 
    - Using `entity_list`

- [ ] `list` https://docs.getodk.org/central-api-entity-management/#entities-metadata
- [ ] `create` https://docs.getodk.org/central-api-entity-management/#creating-an-entity; I think it should take a required entity list name, required label, required dict for data, and an optional uuid (function can generate a uuid4 if none provided)
- [ ] `get_table` https://docs.getodk.org/central-api-odata-endpoints/#id3 same as https://getodk.github.io/pyodk/submissions/#pyodk._endpoints.submissions.SubmissionService.get_table

Should we also create an `entity_list` module by analogy to the `form`/`submission` separation?
- [ ] `list` https://docs.getodk.org/central-api-dataset-management/#datasets
- [ ] `create` upcoming (https://github.com/getodk/central/issues/575)
- [ ] `add_property` upcoming (https://github.com/getodk/central/issues/576)

### PR DESCRIPTION
**Pull request by lindsay-stevens**
Closes #62

Adds methods:
- client.entity_lists.list
- client.entities.list
- client.entities.create
- client.entities.get_table

Other entity_lists methods in #62 not included as they aren't publicly documented yet, so I don't know what the proper call pattern or response type/schema is. Although I took a guess based on manual testing for the test setup function in `tests/utils/entity_lists.py`.

#### What has been done to verify that this works as intended?

E2E tests added to `test_client.py` as well as usual unit tests.

#### Why is this the best possible solution? Were any other approaches considered?

I split entities from entity_lists, because they are separate object types. Due to the similar names it was confusing for me initially to tell the difference between an entity and entity_list and what they are supposed to do. Especially considering `entity.list()`, `entity_list.list()`. So the docstrings for EntityService and EntityListService try to clarify the difference.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Shiny new features!

#### Do we need any specific form for testing your changes? If so, please attach one.

No it's automated in `test_client.py`.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.

The auto-generated docs is a good start but maybe we should add a notebook or something once the extra entity_list methods are available. Something like `test_client.py`/`test_entities__create_and_query` but more elaborate and relatable.

#### Before submitting this PR, please make sure you have:
- [x] included test cases for core behavior and edge cases in `tests`
- [x] run `python -m unittest` and verified all tests pass
- [x] run `ruff format pyxform tests` and `ruff check pyxform tests` to lint code
- [x] verified that any code or assets from external sources are properly credited in comments

### CODE REVIEW

File: pyodk/_endpoints/entities.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+    ```python
+    from pyodk.client import Client
+
+    client = Client()
+    data = client.entities.list()
+    ```
+
+    Conceptually, an Entity's parent object is an EntityList. Each EntityList may
+    have multiple Entities. In Python parlance, EntityLists are like classes, while
```
Comment (lognaturel) [inline pyodk/_endpoints/entities.py:54]: For developers who are very comfortable with the general notion of "an abstract representation" and "an instance of that representation" we've probably obfuscated things. What we've found talking to more typical users, though, is that "a collection of things that are all the same" and "a specific thing in that collection" is much more comfortable. So that's how we ended up where we did.

File: pyodk/_endpoints/entities.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            logger=log,
+        )
+        data = response.json()
+        return [Entity(**r) for r in data]
+
+    def create(
+        self,
+        label: str,
+        data: dict,
+        entity_list_name: str | None = None,
```
Comment (lognaturel) [inline pyodk/_endpoints/entities.py:104]: This should also be required. There is no notion of an entity that lives outside of an entity list.

Comment (lognaturel) [reply pyodk/_endpoints/entities.py:104]: Ah, right, it's because a default can be set. I seem to have trouble remembering that pattern!

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
> entity_list.list()

Yes, that's pretty miserable indeed. Luckily, I believe it will be a rare thing to do. I imagine generally clients will know which entity list to operate on without having to list them. We'll find out! We can always adjust if we need to.

> maybe we should add a notebook or something

Maybe something like https://forum.getodk.org/t/datasets-in-central-are-now-entity-lists-please-help-translate/42893/4?u=ln? I think going through a data source and creating entities one by one will continue to be a relevant automation even when we release bulk entity creation.

File: pyodk/_endpoints/entities.py (hunk 2 above)
Comment (lindsay-stevens) [reply pyodk/_endpoints/entities.py:104]: Yep!

File: pyodk/_endpoints/entities.py (hunk 1 above)
Comment (lindsay-stevens) [reply pyodk/_endpoints/entities.py:54]: Maybe we should stick to examples featuring donkeys and turtles!
